src/test/java/net/sf/jabref/wizard/auximport/AuxSubGeneratorTest.java

```diff
@@ -0,0 +1,105 @@
+package net.sf.jabref.wizard.auximport;
+
+import static org.junit.Assert.*;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import net.sf.jabref.Globals;
+import net.sf.jabref.JabRefPreferences;
+import net.sf.jabref.importer.ParserResult;
+import net.sf.jabref.importer.fileformat.BibtexParser;
+import net.sf.jabref.model.database.BibDatabase;
+
+public class AuxSubGeneratorTest {
+
+    @Before
+    public void setUp() throws Exception {
+        Globals.prefs = JabRefPreferences.getInstance();
+    }
+
+    @Test
+    public void testNormal() {
+        InputStream originalStream = AuxSubGeneratorTest.class.getResourceAsStream("origin.bib");
+        File auxFile = new File(AuxSubGeneratorTest.class.getResource("paper.aux").getFile());
+        try (InputStreamReader originalReader = new InputStreamReader(originalStream)) {
+            ParserResult result = BibtexParser.parse(originalReader);
+
+            AuxSubGenerator auxSubGenerator = new AuxSubGenerator();
+            List<String> unresolved = auxSubGenerator.generate(auxFile.getAbsolutePath(), result.getDatabase());
+            assertFalse(auxSubGenerator.emptyGeneratedDatabase());
+            assertEquals(0, unresolved.size());
+            assertEquals(0, auxSubGenerator.getNotResolvedKeysCount());
+            BibDatabase newDB = auxSubGenerator.getGeneratedDatabase();
+            assertEquals(2, newDB.getEntries().size());
+            assertEquals(2, auxSubGenerator.getResolvedKeysCount());
+            assertEquals(2, auxSubGenerator.getFoundKeysInAux());
+            assertEquals(auxSubGenerator.getFoundKeysInAux(),
+                    auxSubGenerator.getResolvedKeysCount() + auxSubGenerator.getNotResolvedKeysCount());
+            assertEquals(0, auxSubGenerator.getNestedAuxCounter());
+            assertEquals(0, auxSubGenerator.getCrossreferencedEntriesCount());
+
+            // Clear
+
+            auxSubGenerator.clear();
```

Comment:
Why do we need a reusable AuxSubGenerator? It would be easier to maintain a one time usable class with final fields instead of passing them to the `generate` method. Then, the AuxSubGenerator could return a new class like AuxSubGeneratorResult which contains all the methods to query it, e.g., getFoundKeysInAux.